packages/template-retail-react-app/app/pages/social-login-redirect/index.jsx

```diff
@@ -14,19 +14,64 @@ import {
     Text,
     Spinner
 } from '@salesforce/retail-react-app/app/components/shared/ui'
-import {useCustomerType} from '@salesforce/commerce-sdk-react'
+
+// Hooks
 import useNavigation from '@salesforce/retail-react-app/app/hooks/use-navigation'
+import {useAuthHelper, AuthHelpers, useShopperBasketsMutation} from '@salesforce/commerce-sdk-react'
+import {useSearchParams} from '@salesforce/retail-react-app/app/hooks'
+import {useCurrentCustomer} from '@salesforce/retail-react-app/app/hooks/use-current-customer'
+import {getConfig} from '@salesforce/pwa-kit-runtime/utils/ssr-config'
+import {useAppOrigin} from '@salesforce/retail-react-app/app/hooks/use-app-origin'
+import {
+    getSessionJSONItem,
+    clearSessionJSONItem,
+    buildRedirectURI
+} from '@salesforce/retail-react-app/app/utils/utils'
 
 const SocialLoginRedirect = () => {
     const navigate = useNavigation()
-    const {isRegistered} = useCustomerType()
+    const [searchParams] = useSearchParams()
+    const loginIDPUser = useAuthHelper(AuthHelpers.LoginIDPUser)
+    const {data: customer} = useCurrentCustomer()
+    // Build redirectURI from config values
+    const appOrigin = useAppOrigin()
+    const redirectPath = getConfig().app.login.social?.redirectURI || ''
+    const redirectURI = buildRedirectURI(appOrigin, redirectPath)
+
+    const locatedFrom = getSessionJSONItem('returnToPage')
+    const mergeBasket = useShopperBasketsMutation('mergeBasket')
+
+    // Runs after successful 3rd-party IDP authorization, processing query parameters
+    useEffect(() => {
+        if (searchParams.code && searchParams.usid) {
+            loginIDPUser.mutateAsync({
+                code: searchParams.code,
+                redirectURI: redirectURI
+            })
+        }
```

Comment:
Think about inverting this condition so that your code is a little flatter, like: 
```
        if (!searchParams.code || !searchParams.usid) {
            return
        }

        loginIDPUser.mutateAsync({
            code: searchParams.code,
            redirectURI: redirectURI
        })
```
Also.. what happens if that condition doesn't result in a login? Or if someone goes to this page without the params?

Reply:
@bendvc I added error handling in case this condition doesn't result in a login. 

If someone goes to this page without the params they will not be redirected to the secure account page as this login API will not be called.